aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,8 +1,315 @@
+
+Defining and Running Tasks in a Series
+--------------------------------------
+When you run multiple tasks, the tasks run concurrently by default. However, if you need to run tasks in a specific order, you must specify when each task is complete, as well as which tasks depend on the completion of another task. 
+
+1.	To define a series of tasks to run in order, replace the ``first`` task that you added above in *gulpfile.js* with the following:
+
+.. code-block:: javascript
+
+	gulp.task("series:first", function () {
+		console.log('first task! <-----');
+	});
+	gulp.task("series:second", ["series:first"], function () {
+		console.log('second task! <-----');
+	});
+	gulp.task("series", ["series:first", "series:second"], function () {});
+
+	You now have three tasks: ``series:first``, ``series:second``, and ``series``. The ``series:second`` task includes a second parameter that specifies an array of tasks that must be run and completed before the ``series:second`` task will run.  As specified in the code above, only the ``series:first`` task must be completed before the ``series:second`` task will run.	
+
+2.	Save *gulpfile.js*.
+3.	In **Solution Explorer**, right-click *gulpfile.js* and select **Task Runner Explorer** if it isn’t already open. 
+4.	In **Task Runner Explorer**, right-click **series** and select **Run**.
+
+	.. image:: bundling-and-minification/_static/07-TaskRunner-Series.png 
+ 
+IntelliSense
+------------
+IntelliSense, which provide coding details while you code, helps you write code faster and with fewer errors. Because the gulp tasks are created using JavaScript, you can use IntelliSense to help code. As you work with JavaScript, IntelliSense lists the objects, functions, properties, and parameters that are available based on your current context. You can select a coding option from the pop-up list provided by IntelliSense to complete the code.
+
+	.. image:: bundling-and-minification/_static/08-IntelliSense.png 
+
+	For more information about IntelliSense, see `JavaScript IntelliSense <https://msdn.microsoft.com/en-us/library/bb385682.aspx>`_.
+	
+Development, Staging, and Production Environments
+-------------------------------------------------
+
+When you use gulp to optimize your client-side files for staging and production, the processed files are saved to a local staging and production location. The *_Layout.cshtml* file uses the **environment** tag to provide two different versions of CSS files. One version of CSS files is for development and the other version is for both staging and production. In Visual Studio 2015, when you change the **ASPNET_ENV** environment variable to ``Production``, Visual Studio will build the Web app and link to the minimized CSS files. The following markup shows the **environment** tags containing link tags to the ``Development`` CSS files and the minimized ``Staging, Production`` CSS files.
```

Comment:
_Layout.cshtml is an MVC thing - is there a way to write this section to be framework agnostic? (Or call out that this is MVC.)